ddl/ddl_api.go

```diff
@@ -980,6 +980,91 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 	return errors.Trace(err)
 }
 
+func (d *ddl) CreateView(ctx sessionctx.Context, s *ast.CreateViewStmt) (err error) {
+	ident := ast.Ident{Name: s.ViewName.Name, Schema: s.ViewName.Schema}
+	is := d.GetInformationSchema(ctx)
+	schema, ok := is.SchemaByName(ident.Schema)
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(ident.Schema)
+	}
+	if is.TableExists(ident.Schema, ident.Name) && !s.OrReplace {
+		return infoschema.ErrTableExists.GenWithStackByArgs(ident)
+	}
+	if err = checkTooLongTable(ident.Name); err != nil {
+		return err
+	}
+	viewInfo, cols := buildViewInfoWithTableColumns(ctx, s)
+
+	for _, v := range viewInfo.Cols {
+		if len(v.L) > mysql.MaxColumnNameLength {
+			return ErrTooLongIdent.GenWithStackByArgs(v.L)
+		}
+	}
+
+	for _, v := range cols {
```

Comment:
It seems that we do not need this check, because that `viewInfo.Cols` will be that same as `cols` if we do not define the columnNames of view explicitly.

Reply:
Have you tried this in MySQL?

This is a check for the length of column alias, and if you want to do this check, you'd better move it to the expression-rewrite phase for ColumnName and make its behavior be totally the same as MySQL.
I recommend you to do this work in an individual PR.
[This](https://dev.mysql.com/doc/refman/8.0/en/identifiers.html) may help.